shared/src/main/scala/io/kaitai/struct/translators/CommonMethods.scala

```diff
@@ -4,9 +4,137 @@ import io.kaitai.struct.datatype.DataType
 import io.kaitai.struct.datatype.DataType._
 import io.kaitai.struct.exprlang.Ast
 import io.kaitai.struct.format.Identifier
-import io.kaitai.struct.precompile.TypeMismatchError
+import io.kaitai.struct.precompile.{MethodNotFoundError, MethodNotFoundErrorWithArg, TypeMismatchError, WrongMethodCall}
+
+sealed trait MethodArgType
+object MethodArgType {
+  case object IntArg extends MethodArgType {
+    override def toString = "integer"
+  }
+  case object FloatArg extends MethodArgType {
+    override def toString = "float"
+  }
+  case object StrArg extends MethodArgType {
+    override def toString = "string"
+  }
+  case object BooleanArg extends MethodArgType {
+    override def toString = "boolean"
+  }
+  case object BytesArg extends MethodArgType {
+    override def toString = "bytes array"
+  }
+  case object ArrayArg extends MethodArgType {
+    override def toString = "array"
+  }
+
+  def byDataType(dataType: DataType): Option[MethodArgType] = {
+    dataType match {
+      case _: IntType => Some(IntArg)
+      case _: FloatType => Some(FloatArg)
+      case _: StrType => Some(StrArg)
+      case _: BooleanType => Some(BooleanArg)
+      case _: BytesType => Some(BytesArg)
+      case _: ArrayType => Some(ArrayArg)
+      case _ => None
+    }
+  }
+
+  def isArgAcceptable(actualType: DataType, expectedType: MethodArgType) =
+    byDataType(actualType).getOrElse(false) == expectedType
```

Comment:
I'm a bit shocked that this type mismatch between left and right side of `==` (left side `MethodArgType | Boolean`, right side `MethodArgType`) isn't a compile-time error, but Scala apparently doesn't see a problem in testing different types for equality.

Anyway, I'd rather see this:

```suggestion
  def isArgAcceptable(actualType: DataType, expectedType: MethodArgType) =
    byDataType(actualType).map((t) => t == expectedType).getOrElse(false)
```